### PR TITLE
Adding Scopes for API 800, 1000, 1200 and 1600

### PR DESCRIPTION
**Pull request by yuvirani**
### Description
Adding Scopes for API 800, 1000, 1200 and 1600

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass for Python 2.7+ & 3.4+(`$ tox`).
- [ ] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
  - [ ] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: hpOneView/resources/settings/scopes.py [hunk 1]
```diff
@@ -131,8 +92,7 @@ def update(self, resource, timeout=-1):
 
         """
         headers = {'If-Match': resource.get('eTag', '*')}
-        return self._client.update(resource, timeout=timeout, default_values=self.DEFAULT_VALUES,
-                                   custom_headers=headers)
+        return super(Scopes, self).update(resource, timeout=timeout, default_values=self.DEFAULT_VALUES, custom_headers=headers)
```
Comment (VenkateshRavula) [inline hpOneView/resources/settings/scopes.py:96]: update() method in Resource class don't accept 'default_values' as argument. So don't pass that argument. init() method in Resource class will anyway fetch the default values.

File: hpOneView/resources/settings/scopes.py [hunk 2]
```diff
@@ -168,28 +130,5 @@ def update_resource_assignments(self, id_or_uri, resource_assignments, timeout=-
...
-            timeout: Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation
-                in OneView; it just stops waiting for its completion.
-
-        Returns:
-            dict: Updated resource.
-        """
-        headers = {'Content-Type': 'application/json-patch+json'}
-        return self._client.patch(id_or_uri, operation, path, value, timeout=timeout, custom_headers=headers)
+        uri = self._helper.build_uri(id_or_uri) + "/resource-assignments"
+        return super(Scopes, self).patch_request(uri, resource_assignments, timeout=timeout)
```
Comment (VenkateshRavula) [inline hpOneView/resources/settings/scopes.py:134]: What is "resource_assignments" here..?
Please refer to resource.py file

Comment (yuvirani) [reply hpOneView/resources/settings/scopes.py:134]: This URI supported only till API version 300, "resource-assignments" takes existing resource uri's and updates the scope.

File: hpOneView/resources/settings/scopes.py (hunk 1 above)
Comment (yuvirani) [reply hpOneView/resources/settings/scopes.py:96]: I have updated scopes and test_scopes files according the change you mentioned here.

**Comment by yuvirani** [comment]
> endpoints file is messedup. Please correct it accordingly.
I have fixed it.

**Pull request merged**